provider/azureProvider.js

```diff
@@ -112,16 +92,14 @@ return this.parsedBindings;
     }
   }
 
-  LoginWithServicePrincipal () {
-return new BbPromise((resolve, reject) => {
-      msRestAzure.loginWithServicePrincipalSecret(servicePrincipalClientId, servicePrincipalPassword, servicePrincipalTenantId, (error, credentials) => {
-        if (error) {
-          reject(error);
-        } else {
-          principalCredentials = credentials;
-          resolve(credentials);
-        }
-      });
+  Login() {
+    return login({ interactiveLoginHandler: (code) => {
+        this.serverless.cli.log(`Paste this code (copied to your clipboard) into the launched browser, and complete the authentication process: ${code}`);
```

Comment:
Can you add more explicit logging to indicate that this is going to create a service principal behind the scenes as new Azure customers will not be familiar with SP user.